### PR TITLE
ffdhe: add option to set default socket timeout

### PR DESCRIPTION
**Pull request by ueno**
The tests using ffdhe8192 take long time on some CI setup (e.g., with
software emulated GMP) and cause intermittent timeout.  This adds -t
option to control the timeout value from the test invocation.

Signed-off-by: Daiki Ueno <dueno@redhat.com>

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [ ] the changes are also reflected in documentation and code comments
- [ ] all new and existing tests pass (see CI results)
- [ ] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [ ] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [x] new and modified scripts were ran against popular TLS implementations:
  - [ ] OpenSSL
  - [ ] NSS
  - [x] GnuTLS
- [ ] required version of tlslite-ng updated in requirements.txt and README.md

Confirmed at:
https://gitlab.com/dueno/gnutls/-/jobs/1903136178

---
This change is [Reviewable](https://reviewable.io/reviews/tlsfuzzer/tlsfuzzer/782)

### CODE REVIEW

File: scripts/test-ffdhe-expected-params.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 import getopt
 from itertools import chain
 from random import sample
+import socket
```
Comment (tomato42) [inline scripts/test-ffdhe-expected-params.py:12]: unused import

File: scripts/test-ffdhe-expected-params.py [hunk 2]
```diff
@@ -67,9 +70,10 @@ def main():
     last_exp_tmp = None
     group_names = []
     dh_file = None
+    timeout = 1.0
```
Comment (tomato42) [inline scripts/test-ffdhe-expected-params.py:73]: shouldn't we keep the default at 5 seconds?

File: scripts/test-ffdhe-negotiation.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
 import getopt
 from itertools import chain
 from random import sample
+import socket
```
Comment (tomato42) [inline scripts/test-ffdhe-negotiation.py:12]: unused import

File: scripts/test-ffdhe-negotiation.py [hunk 4]
```diff
@@ -65,9 +68,10 @@ def main():
     expected_failures = {}
     last_exp_tmp = None
     sig_algs = None  # `sigalgs` w/o underscore is used for client certificates
+    timeout = 1.0
```
Comment (tomato42) [inline scripts/test-ffdhe-negotiation.py:71]: same here, probably 5s would be better

**Review by tomato42 — CHANGES_REQUESTED** [review]

**Comment by ueno** [comment]
@tomato42 I've addressed the issues; could you take a second look?

**Review by tomato42 — APPROVED** [review]

**Comment by tomato42** [comment]
Looks good but will need to fix py3.3 before merge

**Comment by tomato42** [comment]
@ueno I've updated the CI config, it should pass now, could you rebase?

**Comment by tomato42** [comment]
@ueno Now it's fixed: https://github.com/tlsfuzzer/tlsfuzzer/actions/runs/1660011256

**Comment by ueno** [comment]
@tomato42 what else should I do to get this merged?

**Comment by tomato42** [comment]
*sigh* 3.9 has a new release (3.9.10) so we'll need another rebase after #787 is merged...
That 3.3 failure looks like a python crash, not sure what causes it

**Comment by tomato42** [comment]
It looks like the py3.3 crash was caused by gmpy or gmpy2, I've disabled them for now on that python version, please rebase

**Review by tomato42 — APPROVED** [review]

**Comment by tomato42** [comment]
thanks! and sorry for the CI failures

**Pull request merged**